### ISSUE TITLE
Optimiser for CV estimator

### ISSUE THREAD

**Issue by jeremiecoullon** [jeremiecoullon/SGMCMCJax#11]
We could add an optimser for the CV estimator. This could be using [Adam](https://jax.readthedocs.io/en/latest/_modules/jax/experimental/optimizers.html#adam) in JAX's optimizer module, or using Optax. The idea is that there is a good optimiser setup that's easy to use.

Example API:
```python
adam_optimiser = build_adam_optimiser(loglikelihood, logprior, data, batch_size, compile=True)
opt_params, loss_array = adam_optimiser(key, Niters, param_IC)
```
- very similar API to the sgmcmc samplers
- the step size could have a default value (`1e-3` is usually good). But then it wouldn't be the first argument as in the sgmcmc function (example: `build_sgld`), so maybe not having it as default it better.
- `compile` would by default be set to `True`, but the optimiser would be a python loop if `compile=False`
- the optimiser returns the optimal parameters as well as an array of loss values throughout the optimisation

On the other hand, perhaps this could be considered out of the scope of the library?

**Comment by jeremiecoullon** [comment]
Added this.
